File: f1-management-frontend/src/services/circuitService.js
import api from './api';
// Lấy tất cả các đường đua
export const getAllCircuits = async () => {
  try {
    const response = await api.get('/circuits');
    return response.data;
  } catch (error) {
    console.error('Error fetching circuits:', error);
    throw error;
  }
};

// Lấy đường đua theo id
export const getCircuitById = async (id) => {
  try {
    const response = await api.get(`/circuits/${id}`);
    return response.data;
  } catch (error) {
    console.error(`Error fetching circuit with id ${id}:`, error);
    throw error;
  }
};

File: race-schedule-service/src/main/java/org/example/racescheduleservice/service/CircuitService.java
package org.example.racescheduleservice.service;

import org.example.racescheduleservice.model.Circuit;
import org.example.racescheduleservice.repository.CircuitRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
// Chỉ cần: lấy tất cả circuit, lấy circuit theo id
@Service
public class CircuitService {

    @Autowired
    private CircuitRepository circuitRepository;

    public List<Circuit> getAllCircuits() {
        return circuitRepository.findAll();
    }

    public Optional<Circuit> getCircuitById(String id) {
        return circuitRepository.findById(id);
    }
}